ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/OpenSSL/tasks/main.yml

```diff
@@ -3,54 +3,80 @@
-  win_unzip:
-    src: C:\temp\openssl-1.1.1.tar.gz
-    dest: C:\temp
-    delete_archive: yes
-  when: (openssl32_installed == False) or (openssl64_installed == False)
+- name: Unpack OpenSSL-1.1.1 for 32-bit installation
+  win_shell: |
+    cd C:\temp
+    C:\7-Zip\7z.exe x C:\temp\openssl-1.1.1.tar.gz
+    C:\7-Zip\7z.exe x C:\temp\openssl-1.1.1.tar
+  when: (openssl32_installed.stat.exists == false)
   tags: openssl
 
-- name: Install OpenSSL-1.1.1 32bit
-  win_shell: |
-    set PATH=C:\Strawberry\perl\bin;C:\openjdk\nasm-2.13.03;%PATH%
-    "C:\Program Files (x86)\Microsoft Visual Studio 12.0\Common7\Tools\vsvars32"
-    perl C:\temp\openssl-1.1.1\Configure VC-WIN32 --prefix=C:\openjdk\OpenSSL-1.1.1-x86_32
-    nmake install
+- name: Install OpenSSL-1.1.1 32-bit
+  win_shell: set PATH=C:\Strawberry\perl\bin;C:\openjdk\nasm-2.13.03;%PATH% && .\vsvars32.bat && cd C:\temp\openssl-1.1.1 && perl C:\temp\openssl-1.1.1\Configure VC-WIN32 --prefix=C:\openjdk\OpenSSL-1.1.1-x86_32 && nmake install > C:\temp\openssl32.log
   args:
+    chdir: 'C:\Program Files (x86)\Microsoft Visual Studio 12.0\Common7\Tools'
     executable: cmd
-  when: (openssl32_installed == False)
+  when: (openssl32_installed.stat.exists == false)
+  tags: openssl
+
+- name: Cleanup OpenSSL 32-bit installation
+  win_file:
+    path: C:\temp\{{ item }}
+    state: absent
+  with_items:
+    - openssl-1.1.1
+    - openssl-1.1.1.tar
+  when: (openssl32_installed.stat.exists == false)
   tags: openssl
 
-- name: Install OpenSSL-1.1.1 64bit
+- name: Unpack OpenSSL-1.1.1 for 64-bit installation
   win_shell: |
-    set PATH=C:\Strawberry\perl\bin;C:\openjdk\nasm-2.13.03;%PATH%
-    "C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\vcvarsall" AMD64
-    perl C:\temp\openssl-1.1.1\Configure VC-WIN64A --prefix=C:\openjdk\OpenSSL-1.1.1-x86_64
-    nmake install
+    cd C:\temp
+    C:\7-Zip\7z.exe x C:\temp\openssl-1.1.1.tar.gz
+    C:\7-Zip\7z.exe x C:\temp\openssl-1.1.1.tar
+  when: (openssl64_installed.stat.exists == false)
+  tags: openssl
+
+- name: Install OpenSSL-1.1.1 64-bit
+  win_shell: set PATH=C:\Strawberry\perl\bin;C:\openjdk\nasm-2.13.03;%PATH% && .\vcvarsall.bat AMD64 && cd C:\temp\openssl-1.1.1 && perl C:\temp\openssl-1.1.1\Configure VC-WIN64A --prefix=C:\openjdk\OpenSSL-1.1.1-x86_64 && nmake install > C:\temp\openssl64.log
   args:
+    chdir: 'C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC'
     executable: cmd
-  when: (openssl64_installed == False)
+  when: (openssl64_installed.stat.exists == false)
+  tags: openssl
+
+- name: Cleanup OpenSSL 64bit installation
+  win_file:
+    path: C:\temp\{{ item }}
+    state: absent
+  with_items:
+    - openssl-1.1.1
+    - openssl-1.1.1.tar
+  when: (openssl64_installed.stat.exists == false)
   tags: openssl
 
-- name: Cleanup OpenSSL
+- name: Cleanup OpenSSL source files
   win_file:
-    path: C:\temp\openssl-1.1.1
+    path: C:\temp\{{ item }}
     state: absent
+  with_items:
+    - openssl-1.1.1.tar.gz
```

Comment:
Relatively minor nit but can we avoid untarring twice by leaving the .tar file from the first `7z x` between 32-bit and 64-it installs, and cleaning it up at the end?

Reply:
Good catch.  Additionally, is there a configure or make option to reset or clean such that the unpacking step is only done once.

Reply:
The new changes I made only extract the files once and clean them all at the end. The `openssl-1.1.1` directory is cleaned by adding the `nmake -f makefile clean` to be used for both installations

Reply:
Looks good, thanks for simplifying.